modules/apps/portal-search/portal-search-admin-web/src/main/resources/META-INF/resources/js/IndexActions.js

```diff
@@ -0,0 +1,628 @@
+			.then(() => {
+				if (data.id !== 'spellCheckDictionaries') {
+					_handleBackgroundTaskStart();
+				}
+				else {
+					setBackgroundTaskMap({});
+
+					_handleSyncIconRemove();
+
+					openToast({
+						message: Liferay.Language.get(
+							'reindexing-finished-successfully'
+						),
+						type: 'success',
+					});
+				}
+			})
+			.catch(() => {
+				setBackgroundTaskMap({});
+
+				_handleSyncIconRemove();
+
+				openToast({
+					message: Liferay.Language.get(
+						'an-unexpected-error-occurred'
+					),
+					type: 'danger',
+				});
+			});
+	};
+
+	/*
+	 * Appends sync icon to the control menu, if not already there.
+	 * Called after the reindex fetch.
+	 */
+	const _handleSyncIconAppend = () => {
+		const currentControlMenu = document.getElementById(
+			`${portletNamespace}controlMenu`
+		);
+
+		const currentControlMenuCategory = currentControlMenu.querySelector(
+			`.${controlMenuCategoryKey}-control-group .control-menu-nav`
+		);
+
+		if (
+			!currentControlMenuCategory.querySelector(
+				`#${portletNamespace}syncIcon`
+			)
+		) {
+			const syncIcon = document.createElement('div');
+
+			syncIcon.className = PORTAL_TOOLTIP_TRIGGER_CLASS;
+			syncIcon.setAttribute(
+				'data-title',
+				Liferay.Language.get('the-portal-is-currently-reindexing')
+			);
+			syncIcon.id = `${portletNamespace}syncIcon`;
+			syncIcon.innerHTML = `
+			<svg class="lexicon-icon" focusable="false">
+				<use href="${Liferay.Icons.spritemap}#reload" />
+			</svg>`;
+
+			currentControlMenuCategory.appendChild(syncIcon);
+		}
+	};
+
+	/*
+	 * Removes sync icon to the control menu, if not already removed.
+	 * Called after the background tasks complete.
+	 */
+	const _handleSyncIconRemove = () => {
+		const currentControlMenu = document.getElementById(
+			`${portletNamespace}controlMenu`
+		);
+
+		const syncIcon = currentControlMenu.querySelector(
+			`.${controlMenuCategoryKey}-control-group .control-menu-nav #${portletNamespace}syncIcon`
+		);
+
+		if (syncIcon?.parentNode) {
+			syncIcon.parentNode.removeChild(syncIcon);
+		}
+	};
+
+	/*
+	 * Returns true if any background task is running for the list of indexers.
+	 * If no list is provided, returns true if there are any background tasks
+	 * running.
+	 */
+	const _isBackgroundTaskRunning = (indexers = []) => {
+		if (indexers.length) {
+			return indexers.some(
+				(type) => typeof backgroundTaskMap[type] !== 'undefined'
+			);
+		}
+
+		return !!Object.keys(backgroundTaskMap).length;
+	};
+
+	return (
+		<ClayLayout.Container
+			className="search-admin-index-actions-container"
+			fluidSize="xl"
+			formSize="lg"
+			id={`${portletNamespace}adminSearchAdminIndexActionsPanel`}
+		>
+			<ClayLayout.ContainerFluid view>
+				<ClayLayout.Row>
+					<ClayLayout.Col size={4}>
+						<ExecutionOptions
+							companyIds={selectedCompanyIds}
+							executionMode={executionMode}
+							executionScope={executionScope}
+							isConcurrentModeSupported={
```

Comment:
With boolean prop names we usually drop the `is`. For example, `disabled`, `active`, `selected`.